backend/graminsta/post/views.py

```diff
@@ -110,14 +114,69 @@ class TimelineView(APIView):
     def get(request):
         """Gets the given user's timeline
 
+
         Parameters
         ----------
         request: GET request
 
         Returns
+
         -------
         response: json format Posts that should be
             displayed at the given user's timeline
         """
         posts = get_timeline_posts(request.user)
         return Response(PostSerializer(posts, many=True).data)
+
+
+class PersonalGalleryView(APIView):
+    """
+    A class based view for viewing all personal posts
+    """
+
+    def get(self, request):
+        """
+        Get all personal post
+
+        Parameters
+        ----------
+        request: GET request
+
+        Returns
+
+        ----------
+        response: list
+            List of all personal posts
+        """
+        posts = get_all_personal_post(request.user)
+        result = []
+        for post in posts:
+            result.append(PostSerializer(post).data)
+        return Response(result)
+
+
+class PersonalCountView(APIView):
```

Comment:
As this is only for personal profile page, how about just inline the data in the response of `PersonalGalleryView`.

```
{
  posts: [post0, post1, post2, ...]
  post_count: 123,
  following_count: 456,
  fan_count: 789
}
```